File: 257-binary-tree-paths/257-binary-tree-paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def dfs(self, root, l, s):
        if not root.left and not root.right:
            l.append(s)
            return 
        if root.left:
            self.dfs(root.left, l, s+"->"+str(root.left.val))
        if root.right:
            self.dfs(root.right, l, s+"->"+str(root.right.val))
    
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        if not root:
            return []
        r = []
        self.dfs(root, r, str(root.val))
        return r

File: 257-binary-tree-paths/257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* root, vector<string>& v, string t)
    {
        if(!root->left && !root->right) 
        {
            v.push_back(t);
            return ;
        }
        if(root->left) dfs(root->left, v, t+"->"+to_string(root->left->val));
        if(root->right) dfs(root->right, v,t+"->"+to_string(root->right->val));
        
    }
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> v;
        if(!root) return {};
        dfs(root, v, to_string(root->val));
        return v;
    }
};